addons/mass_mailing/static/src/js/mass_mailing_html_field.js

```diff
@@ -25,10 +25,18 @@ export class MassMailingHtmlField extends HtmlField {
 
     setup() {
         super.setup();
+        // Use reactive instead of useState because currently, this component
+        // does not need to re-render when its state changes. If it does, change
+        // to `useState`.
+        this.state = reactive({
+            selectedTheme: null,
+            $scrollable: null,
+        });
```

Comment:
> The reason it is not a state transferred as a prop is because you want to give it to a sub-sub-child, right? Or is there another reason?

Yes, that's the reason, to not pollute the wysiwyg props with a prop it will not use.

> The only code using this reactive object is addons/mass_mailing/static/src/js/snippets.editor.js, right? Actually does it even need to be reactive here? Is not the use of this.mailingState = useState(this.env.mailingState); enough?

Yes and no, only snippets.editor.js will read the variable, but it needs to be reactive before that so that setting something in that object triggers the reactive chain (re-render => useEffect)

In your example, doing `this.envConfig.selectedTheme = newTheme` would do nothing, as that object is a plain object, and not the reactive proxy. (Whenever an object is already reactive, using "useState" will subscribe the component's render method to the reactive object, it will not re-create a reactive object, so SnippetsMenu and MassMailingHtmlField share the same reactive object and can trigger stuff that way.)

> I am confused by this: "component does not need to re-render so no useState"... is it worth a comment? Should not you just not call it state?

I had a feeling that it could be confusing, that is why I tried to comment: I called it state because it's a reactive value that will cause a child component to re-render. I do not use `useState` because, from what I understand, setting a value in a state set by useState will cause OWL to re-render the component (I think at least once, but I might be wrong), which is not necessary here as the value is not really used by any OWL feature.

I do not mind calling it something else, but it would be nice to have a name that indicates that child components (only SnippetsMenu but any child component can subscribe to that reactive value) will react if that thing is accessed.

I'm okay with fieldConfig or fieldState, whichever you prefer.

Reply:
> In your example, doing this.envConfig.selectedTheme = newTheme would do nothing, as that object is a plain object, and not the reactive proxy.

Oh, ok, I see. I imagined that `reactive(obj) == obj`, makes sense that's not true.

> by useState will cause OWL to re-render the component (I think at least once, but I might be wrong)

I would imagine that's not true if the state is not used by the component XML definition. But not using `useState` seems good here given the fact it's a reactive stuff passed through env.

> I do not mind calling it something else

Not sure. I'll ask an OWL expert. I prepared a schema to do so:
![image](https://github.com/odoo/odoo/assets/10338094/d38e6c16-3dc6-4917-b783-0e203ac0abce)

Reply:
EDIT, réponse de GED:
> en général, j'aime bien un nom qui représente le concept. du genre notification et pas notificationState. ou overlay, ou model

So I would avoid `state`, let's go with `fieldConfig`. `fieldReactiveConfig` if you prefer maybe :shrug: